Yosemite/Yosemite/PointOfSale/POSItemProvider.swift

```diff
@@ -0,0 +1,7 @@
+public protocol POSItem {
+    var itemID: UUID { get }
+}
```

Comment:
Design this to match the designs. If we get this right, there's no need to specialise to products when we're building the view.
```suggestion
public protocol POSItem {
    var itemID: UUID { get }
    var name: String { get }
    var price: String { get } // It's arguable whether this should be pre-formatted in Yosemite or done in the view.
    var imageURL: URL { get } // This is not an image, so we can lazy load the image in the view. There are probably other ways
    var details: [String] { get } // A very simple way to do category/subcategory strings. We'd probably grow out of it, and an array of enum cases might give more flexibility. We can still have specific named properties for these on the concrete implementations though, and make this a computed var which puts together the named properties in the right order.
    func makeCartItem() -> POSCartItem // Called to make something when it's time to add to the cart.
}
```

Reply:
Oh! I didn't know this. I used `itemID` just as an initial property to get the ball rolling with the protocol, and didn't realize that the miss-match here is what was causing the casting problem in SwiftUI. I updated the model here a2469ec73e2521e03ff58a96d87bd23539cb5611 so we can remove the downcast and will update the property names and adjust the model on different commits.